File: jumping_on_the_clouds.py
def jumpingOnClouds(c):
    jumps, i = 0, 0
    while i < len(c)-1:
        # if 2 cloud jumps
        if i+2 < len(c) and c[i+2] != 1:
            i += 1
        jumps += 1
        i += 1
    print(jumps)

jumpingOnClouds([0, 0, 0, 1, 0, 0]) # 3
jumpingOnClouds([0, 0, 1, 0, 0, 1, 0]) # 4
jumpingOnClouds([0, 1, 0, 0, 0, 1, 0]) # 3
jumpingOnClouds([0, 0, 1, 0, 0, 0, 0, 1, 0, 0]) #6
jumpingOnClouds([0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0 ,0])

File: javascript/warmup/jumping_on_the_clouds.js
jumpingOnClouds = c =>{
    let jumps  = 0;
    let i = 0;

    while(i < c.length - 1 ){
        if((i+2) < c.length && c[i+2] != 1){
            i += 1;
        }

        jumps += 1;
        i += 1;
    }

    console.log(jumps);
    return jumps;
}

jumpingOnClouds([0, 0, 0, 1, 0, 0]) //  3
jumpingOnClouds([0, 0, 1, 0, 0, 1, 0]) //  4
jumpingOnClouds([0, 1, 0, 0, 0, 1, 0]) //  3
jumpingOnClouds([0, 0, 1, 0, 0, 0, 0, 1, 0, 0]) // 6